app/bin/server_common.dart

```diff
@@ -89,7 +89,32 @@ Future<String> obtainServiceAccountEmail() async {
   return response.body.trim();
 }
 
-Future<DatastoreDB> initializeApiaryDatastore() async {
+/// Wrapper function used to work around issue with running gRPC Datastore on
+/// MacOS.
+///
+/// On
+///    * Linux this function simply returns `fun(dbService)`
+///
+///    * MacOS this function makes a new service scope, optionally builds an
+///      apiary Datastore and registers an apiary Datastore in the new service
+///      scope
+///
+Future withCorrectDatastore(Future fun()) {
+  if (Platform.isMacOS) {
+    return fork(() async {
+      _apiaryDatastore ??= await _initializeApiaryDatastore();
```

Comment:
Consider resetting `_apiaryDatastore` after `await fun()`. So it's nesting, but the outermost one, resets it to `null`.